nbs/03_matcher.ipynb

```diff
@@ -10,7 +10,7 @@
   },
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #2.    def _attributes_match(input_graph_attrs: dict, pattern_attrs: dict) -&gt; bool:
</pre><p>this function name is an abstraction leak. you know that these dicts are attributes of nodes, but the function doesn't need to know that. it simply checks that x is a sub dict of y</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/DeanLight/graph_rewrite/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Reply:
<p>I agree that it is an abstraction leak - I tried to name it similarly to an earlier function - _attributes_exist.</p><p>I see that I later commented out the usage of that function, so it was not needed in the end - I will delete it.</p>